### PR TITLE
[Fixes #7692] Layers caching problem

### LINKED ISSUE #7692: Layers caching problem
The layer panes are set to 512 by default but the configuration for the cache in GWC is not active and the following dridsets: EPSG: 900913x2 and EPSG: 4326x2 should be set by default when a layer is loaded

### PR DESCRIPTION
**Pull request by mattiagiupponi**
This PR will fix an issue regarding the cache of the layer for users with geo-limits.
Instead calling `toggle_layer_cache` for each user, not will call the method only if at least of the user have geo-litmits enabled.
- Add test for new function `TestGetUserGeolimits`
- Fix tests
- Removing set of `gwc_gridSubsets` (no longer required with the new geoserver versions

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [ ] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [ ] The issue connected to the PR must have Labels and Milestone assigned
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [x] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

File: geonode/security/utils.py [hunk 1]
```diff
@@ -352,25 +352,14 @@ def toggle_layer_cache(layer_name, enable=True, filters=None, formats=None):
                 gwc_enabled.text = str(enable).lower()
 
                 gwc_mimeFormats = tree.find('mimeFormats')
-                if formats is None:
-                    tree.remove(gwc_mimeFormats)
-                else:
+                tree.remove(gwc_mimeFormats)
```
Comment (afabiani) [inline geonode/security/utils.py:355]: Should we probably check if the element `gwc_mimeFormats` exists?

**Review by afabiani — CHANGES_REQUESTED** [review]
1. The title of the PR should be in the form: [Fixes #issue] <title issue>
2. There are some tests failing

```
======================================================================
ERROR: test_set_unset_user_layer_permissions (geonode.people.tests.TestSetUnsetUserLayerPermissions)
Test that user permissions are set for layers
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/django/test/utils.py", line 387, in inner
    return func(*args, **kwargs)
  File "/usr/src/geonode/geonode/people/tests.py", line 77, in test_set_unset_user_layer_permissions
    utils.set_layers_permissions(
  File "/usr/src/geonode/geonode/layers/utils.py", line 1062, in set_layers_permissions
    resource.set_permissions(perm_spec)
  File "/usr/local/lib/python3.8/contextlib.py", line 75, in inner
    return func(*args, **kwds)
  File "/usr/src/geonode/geonode/security/models.py", line 318, in set_permissions
    _, _, _disable_layer_cache, _, _, _ = get_user_geolimits(self.layer, _user, group_perms, gf_services)
  File "/usr/src/geonode/geonode/security/utils.py", line 625, in get_user_geolimits
    _group = group if isinstance(group, str) else group.name
AttributeError: 'dict' object has no attribute 'name'

======================================================================
ERROR: test_unset_group_layer_perms (geonode.people.tests.TestSetUnsetUserLayerPermissions)
Test that group permissions are unset for layers
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/django/test/utils.py", line 387, in inner
    return func(*args, **kwargs)
  File "/usr/src/geonode/geonode/people/tests.py", line 139, in test_unset_group_layer_perms
    utils.set_layers_permissions(
  File "/usr/src/geonode/geonode/layers/utils.py", line 1062, in set_layers_permissions
    resource.set_permissions(perm_spec)
  File "/usr/local/lib/python3.8/contextlib.py", line 75, in inner
    return func(*args, **kwds)
  File "/usr/src/geonode/geonode/security/models.py", line 318, in set_permissions
    _, _, _disable_layer_cache, _, _, _ = get_user_geolimits(self.layer, _user, group_perms, gf_services)
  File "/usr/src/geonode/geonode/security/utils.py", line 625, in get_user_geolimits
    _group = group if isinstance(group, str) else group.name
AttributeError: 'dict' object has no attribute 'name'

======================================================================
ERROR: test_get_arcgis_alternative_structure (geonode.services.tests.ModuleFunctionsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/unittest/mock.py", line 1325, in patched
    return func(*newargs, **newkeywargs)
  File "/usr/src/geonode/geonode/services/tests.py", line 491, in test_get_arcgis_alternative_structure
    result = handler.create_geonode_service(test_user)
  File "/usr/src/geonode/geonode/services/serviceprocessors/arcgis.py", line 114, in create_geonode_service
    version=self.parsed_service._json_struct.get("currentVersion", 0.0),
  File "/usr/local/lib/python3.8/site-packages/arcrest/server.py", line 266, in _json_struct
    if self._contents is not Ellipsis:
  File "/usr/local/lib/python3.8/site-packages/arcrest/server.py", line 243, in _contents
    handle = compat.urllib2.urlopen(request)
  File "/usr/local/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/local/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/usr/local/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/usr/local/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/local/lib/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found

======================================================================
FAIL: test_layer_get_list_layer_private_to_one_user (geonode.api.tests.PermissionsApiTests)
Test that if a layer is only visible by admin, then does not appear
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/src/geonode/geonode/api/tests.py", line 114, in test_layer_get_list_layer_private_to_one_user
    self.assertEqual(len(self.deserialize(resp)['objects']), 8)
AssertionError: 7 != 8

======================================================================
FAIL: test_layers_get_list_auth_some_public (geonode.api.tests.PermissionsApiTests)
Test that if a layer is not public then all are returned if the
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/src/geonode/geonode/api/tests.py", line 94, in test_layers_get_list_auth_some_public
    self.assertEqual(len(self.deserialize(resp)['objects']), 8)
AssertionError: 7 != 8

======================================================================
FAIL: test_should_disable_cache_for_anonymous_with_geolimits (geonode.security.tests.TestGetUserGeolimits)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/src/geonode/geonode/security/tests.py", line 1871, in test_should_disable_cache_for_anonymous_with_geolimits
    self.assertTrue(_disable_layer_cache)
AssertionError: False is not true

======================================================================
FAIL: test_layers_locations_composition_map_default_bbox (geonode.thumbs.tests.test_unit.ThumbnailsUnitTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/src/geonode/geonode/thumbs/tests/test_unit.py", line 193, in test_layers_locations_composition_map_default_bbox
    self.assertEqual(bbox, expected_bbox, "Expected calculated BBOX to match pre-converted one.")
AssertionError: Lists differ: [-184[27 chars]31394347, -20040289.59992574, 16329038.485056708, 'EPSG:3857'] != [-184[27 chars]31394347, -20040289.599925745, 16329038.485056704, 'EPSG:3857']

First differing element 2:
-20040289.59992574
-20040289.599925745

  [-18411664.521739896,
   1414810.0631394347,
-  -20040289.59992574,
+  -20040289.599925745,
?                    +

-  16329038.485056708,
?                   ^

+  16329038.485056704,
?                   ^

   'EPSG:3857'] : Expected calculated BBOX to match pre-converted one.

======================================================================
FAIL: test_layers_locations_layer_default_bbox (geonode.thumbs.tests.test_unit.ThumbnailsUnitTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/src/geonode/geonode/thumbs/tests/test_unit.py", line 143, in test_layers_locations_layer_default_bbox
    self.assertEqual(bbox, expected_bbox, "Expected calculated BBOX to match pre-converted one.")
AssertionError: Lists differ: [-823[24 chars].787127878, 4969844.155936863, 4984363.9488296695, 'epsg:3857'] != [-823[24 chars].787127878, 4969844.155936863, 4984363.948829668, 'epsg:3857']

First differing element 3:
4984363.9488296695
4984363.948829668

  [-8238681.428369759,
   -8220320.787127878,
   4969844.155936863,
-  4984363.9488296695,
?                  ^^

+  4984363.948829668,
?                  ^

   'epsg:3857'] : Expected calculated BBOX to match pre-converted one.

======================================================================
FAIL: test_layers_locations_simple_map_default_bbox (geonode.thumbs.tests.test_unit.ThumbnailsUnitTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/src/geonode/geonode/thumbs/tests/test_unit.py", line 173, in test_layers_locations_simple_map_default_bbox
    self.assertEqual(bbox, expected_bbox, "Expected calculated BBOX to match pre-converted one.")
AssertionError: Lists differ: [-823[24 chars].787127878, 4969844.155936863, 4984363.9488296695, 'epsg:3857'] != [-823[24 chars].787127878, 4969844.155936863, 4984363.948829668, 'epsg:3857']

First differing element 3:
4984363.9488296695
4984363.948829668

  [-8238681.428369759,
   -8220320.787127878,
   4969844.155936863,
-  4984363.9488296695,
?                  ^^

+  4984363.948829668,
?                  ^

   'epsg:3857'] : Expected calculated BBOX to match pre-converted one.

----------------------------------------------------------------------
Ran 401 tests in 6791.481s

FAILED (failures=6, errors=3)
```

**Comment by mattiagiupponi** [comment]
> There are some tests failing

3 tests are not strictly related to this PR, but I'll take a look at them. 
The ones that are not related are the following 3:
- `test_get_arcgis_alternative_structure (geonode.services.tests.ModuleFunctionsTestCase)`
  - It fails because the URL used as integration tests is no longer available 

----

- `test_layers_locations_layer_default_bbox (geonode.thumbs.tests.test_unit.ThumbnailsUnitTest)`
- `test_layers_locations_simple_map_default_bbox (geonode.thumbs.tests.test_unit.ThumbnailsUnitTest)`

The issue for those two tests comes from the upgrade of the library `pyproj` from `3.0.1` to `3.1.0`. In this function
https://github.com/GeoNode/geonode/blob/523137d56b7131733626199bbee8034f8bee6b6f/geonode/thumbs/utils.py#L71-L76

The Object `Transformer` is used to transform the box, probably there are some differences between the two versions, which is causing the TestFaling.

@afabiani 2 solutions:
- Fix the tests to adapt to the expected results
- Downgrade library to version `3.0.1`

**Review by afabiani — APPROVED** [review]

**Comment by afabiani** [comment]
Failing tests not related to the PR. To be fixed as part of another PR.

**Pull request merged**